templates/filebeat.yml

```diff
@@ -13,6 +13,12 @@ filebeat:
       scan_frequency: 10s
       harvester_buffer_size: {{ harvester_buffer_size }}
       max_bytes: {{ max_bytes }}
+      {% if fields -%}
+      fields: 
+        {% for fieldvalue in fields -%}
+        {{ fieldvalue.split(':')[0] }}: {{ fieldvalue.split(':')[1] }}
```

Comment:
This looks fine at first glance, however its relying on there being 2 elements. If someone fat-fingers their value this will cause the template to malfunction with this current logic. So its a bit more resilient, i would probably change the second value from `{{ fieldvalue.split(':')[1]}}` to `{{ fieldvalue.split(':')[-1]}}`  so it defaults to whatever the 'last' value that came from the split operation, even if its the same as the suspected key, which is errant, but wont throw a stack if the `:` is omitted.

I will agree, this is a deficiency in how juju doesn't superimpose any client-side validation, so we as layer authors have to be vigilant in trying to safeguard against this behavior.

That or add a validation routine in the layer code before it gets passed to the template system. Pick your poison :)